app/models/admin/status_batch_action.rb

```diff
@@ -47,6 +47,12 @@ def handle_delete!
       statuses.each do |status|
         status.discard_with_reblogs
         log_action(:destroy, status)
+
+        next unless status.with_media?
+
+        # Immediately remove public copy of media instead of waiting for
+        # the vacuum_orphaned_records job to take care of it later on
+        Admin::MediaAttachmentDeletionWorker.perform_inline(status.media_attachments)
```

Comment:
Should this function be `perform()` instead of `perform_inline()`?  I don't see `perform_inline()` defined in the class, but maybe it's some sort of Rails sorcery?

Reply:
Thanks Amri for having a look! We want the action to trigger immediately, so it should be `perform_inline`. The worker class itself has one method - `perform`, and it can be invoked wit `perform_inline` for sync action or `perform_async` for asynchronous action behind the scenes. 

There are a couple more methods like `perform_in` but they delay action rather than speed it up, so `perform_inline` is the best we can access, with the caveat that it will hold up that entire transaction though. 

https://www.rubydoc.info/gems/sidekiq/Sidekiq/Job/ClassMethods

Reply:
Ah – thanks for the education!